### PR TITLE
Properly validate the schema

### PR DESCRIPTION
**Pull request by sergiusens**
There's a refactor to not duplicate work and to also move stuff out of
`snapcraft.yaml`

LP: #1564607

Signed-off-by: Sergio Schvezov <sergio.schvezov@ubuntu.com>

### CODE REVIEW

**Comment by sergiusens** [comment]
retest this please

**Comment by sergiusens** [comment]
```
integrationtests     FAIL non-zero exit status 1
examplestests        FAIL non-zero exit status 1
```

**Comment by sergiusens** [comment]
retest this please

**Comment by sergiusens** [comment]
@elopio can you figure out the problem with the autopackage tests?

**Comment by elopio** [comment]
retest this please

File: snapcraft/yaml.py [hunk 1]
```diff
@@ -486,26 +478,23 @@ def _create_pkg_config_override(bindir, installdir, stagedir):
     return ['PATH={}:$PATH'.format(bindir)]
 
 
-def _validate_snapcraft_yaml(snapcraft_yaml):
-    schema_file = os.path.abspath(os.path.join(common.get_schemadir(),
-                                               'snapcraft.yaml'))
+def _expand_filesets_for(stage, properties):
```
Comment (kyrofa) [inline snapcraft/yaml.py]: Consider s/stage/step/ ?

**Comment by sergiusens** [comment]
@kyrofa fixed and also fixed a test

**Pull request merged**